internal/docker/deployment.go

```diff
@@ -99,6 +99,7 @@ func (d *Deployment) Register(t ct.TestLike, hsName string, opts helpers.Registr
 		Client:           client.NewLoggedClient(t, hsName, nil),
 		SyncUntilTimeout: 5 * time.Second,
 		Debug:            d.Deployer.debugLogging,
+		Password:         opts.Password,
```

Comment:
The password, if empty and using the default, is not written back to the client [below](https://github.com/matrix-org/complement/blob/d690a2ff37d1a62728172ddca90229493236503c/internal/docker/deployment.go#L108-L111).